src/ui/layout/tutorial.md

```diff
@@ -13,183 +13,275 @@ diff2html: true
-You'll build the layout for the following app:
+This guide explains how to build layouts in Flutter.
 
-{% include docs/app-figure.md img-class="site-mobile-screenshot border"
-    image="ui/layout/lakes.jpg" caption="The finished app" %}
+The example code builds the layout for the following app:
 
-This guide then takes a step back to explain Flutter's
-approach to layout, and shows how to place a single widget
-on the screen. After a discussion of how to lay widgets
-out horizontally and vertically, some of the most common
-layout widgets are covered.
+{% include docs/app-figure.md
+   img-class="site-mobile-screenshot border"
+   image="ui/layout/layout-demo-app.png"
+   caption="The finished app"
+   width="50%" %}
 
-If you want a "big picture" understanding of the layout mechanism,
-start with [Flutter's approach to layout][].
+**Credits:** Photo by [Dino Reichmuth][ch-photo] on [Unsplash][].
+Text by [Switzerland Tourism][].
 
-## Step 0: Create the app base code
+If you want a high-level understanding of the layout mechanism,
+start with [Flutter's approach to layout][].
 
-Make sure to [set up][] your environment,
-then do the following:
+## Create the app base code
 
 <?code-excerpt path-base="layout/base"?>
 
- 1. [Create a new Flutter app][new-flutter-app].
- 2. Replace the contents in `lib/main.dart` with the following code:
+1. [Set up your Flutter environment][].
 
-    <?code-excerpt "lib/main.dart (all)" title?>
-    ```dart
-    import 'package:flutter/material.dart';
+1. [Create a new Flutter app][new-flutter-app].
 
-    void main() => runApp(const MyApp());
+1. Replace the contents of `lib/main.dart` with the following code.
+   This app uses a parameter for the app title and the title shown
+   on the app's `appBar`. This decision was made to simplify the code.
 
-    class MyApp extends StatelessWidget {
-      const MyApp({super.key});
+   <?code-excerpt "lib/main.dart (all)" title?>
+   ```dart
+   import 'package:flutter/material.dart';
 
-      @override
-      Widget build(BuildContext context) {
-        return MaterialApp(
-          title: 'Flutter layout demo',
-          home: Scaffold(
-            appBar: AppBar(
-              title: const Text('Flutter layout demo'),
-            ),
-            body: const Center(
-              child: Text('Hello World'),
-            ),
-          ),
-        );
-      }
-    }
-    ```
+   void main() => runApp(const MyApp());
+
+   class MyApp extends StatelessWidget {
+     const MyApp({super.key});
+
+     @override
+     Widget build(BuildContext context) {
+       const String appTitle = 'Flutter layout demo';
+       return MaterialApp(
+         title: appTitle,
+         home: Scaffold(
+           appBar: AppBar(
+             title: const Text(appTitle),
+           ),
+           body: const Center(
+             child: Text('Hello World'),
+           ),
+         ),
+       );
+     }
+   }
+   ```
+
+## Diagram the layout
```

Comment:
While I do prefer this updated version, I've always found this section of the page to be superfluous. Are readers really going to break out a pad of paper and draw a layout?  

I can see an argument for why it's important, but I also think the truly valuable information here is in the annotated images, which could just be integrated into the steps themselves (starting with `## Add the title screen widget`

Reply:
Yeah, I see that point and debated removing it myself, but getting the developer to stop and think about design before coding didn't see like a bad idea.